app/backend/src/couchers/models.py

```diff
@@ -654,3 +654,324 @@ class InitiatedUpload(Base):
+    """
+
+    __tablename__ = "clusters"
+
+    id = Column(BigInteger, primary_key=True)
+    name = Column(String, nullable=False)
+    main_page_id = Column(ForeignKey("pages.id"), nullable=False, unique=True, index=True)
+    created = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
+
+    main_page = relationship("Page", backref="cluster", uselist=False, foreign_keys="Cluster.main_page_id")
+
+    nodes = relationship("Cluster", backref="clusters", secondary="node_cluster_associations")
+    pages = relationship("Page", backref="clusters", secondary="cluster_page_associations")
+    events = relationship("Event", backref="clusters", secondary="cluster_event_associations")
+    discussions = relationship("Discussion", backref="clusters", secondary="cluster_discussion_associations")
+    subscribers = relationship("User", backref="clusters", secondary="cluster_subscriptions")
+
+
+class NodeClusterAssociation(Base):
+    """
+    NodeClusterAssociation, grouping of nodes
+    """
+
+    __tablename__ = "node_cluster_associations"
+    __table_args__ = (UniqueConstraint("node_id", "cluster_id"),)
+
+    id = Column(BigInteger, primary_key=True)
+
+    node_id = Column(ForeignKey("nodes.id"), nullable=False, index=True)
+    cluster_id = Column(ForeignKey("clusters.id"), nullable=False, index=True)
+
+    node = relationship("Node", backref="node_cluster_associations")
+    cluster = relationship("Cluster", backref="node_cluster_associations")
+
+
+class GroupRole(enum.Enum):
+    member = 1
+    admin = 2
+
+
+class ClusterSubscription(Base):
+    """
+    ClusterSubscription of a user
+    """
+
+    __tablename__ = "cluster_subscriptions"
+    __table_args__ = (UniqueConstraint("user_id", "cluster_id"),)
+
+    id = Column(BigInteger, primary_key=True)
+
+    user_id = Column(ForeignKey("users.id"), nullable=False, index=True)
+    cluster_id = Column(ForeignKey("clusters.id"), nullable=False, index=True)
+    role = Column(Enum(GroupRole), nullable=False)
+    joined = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
+    left = Column(DateTime(timezone=True), nullable=True)
+
+    user = relationship("User", backref="cluster_subscriptions")
+    cluster = relationship("Cluster", backref="cluster_subscriptions")
+
+
+class ClusterPageAssociation(Base):
+    """
+    pages related to clusters
+    """
+
+    __tablename__ = "cluster_page_associations"
+    __table_args__ = (UniqueConstraint("page_id", "cluster_id"),)
+
+    id = Column(BigInteger, primary_key=True)
+
+    page_id = Column(ForeignKey("pages.id"), nullable=False, index=True)
+    cluster_id = Column(ForeignKey("clusters.id"), nullable=False, index=True)
+
+    page = relationship("Page", backref="cluster_page_associations")
+    cluster = relationship("Cluster", backref="cluster_page_associations")
+
+
+class PageType(enum.Enum):
+    main_page = 1
+    point_of_interest = 2
+    guide = 3
+    event = 4
```

Comment:
No `event` in here, it's a separate table

Reply:
fixed